standard/interfaces.md

```diff
@@ -305,52 +427,132 @@ These rules ensure that any covariant or contravariant usage of the interface re
+
+One override `M1` is considered *more specific* than another override `M2` if `M1` is declared on type `T1`, `M2` is declared on type `T2`, and either
+
+1. `T1` contains `T2` among its direct or indirect interfaces, or
+1. `T2` is an interface type but `T1` is not an interface type.
+
+> *Example*:
+>
+> <!-- Example: {template:"standalone-lib-without-using", name:"InterfaceMethods4", expectedErrors:["CS8705"]} -->
+> ```csharp
+> interface IA
+> {
+>     void M() { Console.WriteLine("IA.M"); }
+> }
+> 
+> interface IB : IA
+> {
+>     void IA.M() { Console.WriteLine("IB.M"); }
+> }
+> 
+> interface IC: IA
+> {
+>     void IA.M() { Console.WriteLine("IC.M"); }
+> }
+> 
+> abstract class C: IB, IC { } // error: no most specific override for 'IA.M'
+> 
+> abstract class D: IA, IB, IC // OK
+> {
+>     public abstract void M();
+> }
+> ```
+>
+> The most specific override rule ensures that a conflict (i.e., an ambiguity arising from diamond inheritance) is resolved explicitly by the programmer at the point where the conflict arises. *end example*
 
-The accessors of an interface property declaration correspond to the accessors of a class property declaration ([§15.7.3](classes.md#1573-accessors)), except that the *accessor_body* shall always be a semicolon. Thus, the accessors simply indicate whether the property is read-write, read-only, or write-only.
+It is an error if in a class declaration the most specific override of some interface method is an abstract override that was declared in an interface.
+
+> *Example*:
+>
+> <!-- Example: {template:"standalone-lib-without-using", name:"InterfaceMethods5", expectedErrors:["CS0535"]} -->
+> ```csharp
+> interface IF
+> {
+>     void M();
+> }
+> 
+> abstract class F : IF { } // error: 'F' does not implement 'IF.M'
+> ```
+>
+> *end example*
+
+### 18.4.3 Interface properties
+
+[§15.7](classes.md#157-properties) and its subclauses cover property declarations in classes. When declaring properties in interfaces, that text needs to be interpreted in the context of an interface, and, where necessary, it is augmented and/or replaced by text in this subclause.
+
+Interface properties are declared using *property_declaration*s ([§15.7.1](classes.md#1571-general)).
+
+*property_modifier* shall not include `override`.
+
+A *property_declaration* that has a *block* as an *accessor_body* is a default implementation ([§18.1](interfaces.md#181-general)), so it is *not* part of the interface’s contract. A *property_declaration* that has no default implementation is always considered part of the interface’s contract; it is *never* considered to be an automatically implemented property ([§15.7.4](classes.md#1574-automatically-implemented-properties)).
```

Comment:
Expression-bodied properties should be considered here. `int Prop => 42;` is a default implementation. Also echoing the concern about excluding DIMs from the "contract."

Reply:
Can we also restrict this to talking about instance properties? I get confused when I enter this paragraph with `static int Prop { get; set; }` in mind, which is an auto prop.